velox/type/Type.h

```diff
@@ -501,6 +537,99 @@ class TypeBase : public Type {
   }
 };
 
+/*
+ * This class represents the Decimal Type to represent the fixed-point numbers
+ * in Velox. The parameter "precision" represents the number of digits the
+ * Decimal Type can support and scale represents the number of digits to the
+ * right of the decimal point.
+ */
+class DecimalType : public Type {
+ public:
+  DecimalType(const uint8_t precision = 18, const uint8_t scale = 0)
+      : Type(
+            precision <= 18 ? TypeKind::SHORT_DECIMAL : TypeKind::LONG_DECIMAL),
+        typmod_(TYPMOD(precision, scale)),
+        isBigDecimal_(precision > 18) {
+    std::stringstream ss;
```

Comment:
We do not need to build this string unless toString() is called. Let's move this code there. We also do not need to store this string as toString() is not used in performance critical code paths.